### ISSUE TITLE
Parse Error: Header overflow

### ISSUE THREAD

**Issue by mikej0hnst0n** [a85/POSTMan-Chrome-Extension#8656]
Since Version 7 of Postman I am experiencing an issue with header size from Some Huawei cloud service APIs. The error manifests itself with Postman as '**Error: Parse Error: Header overflow**'. Deeper inspection using curl shows that there is an element in the response header 'X-Subject-Token' which is just over 8Kbytes in length. This authorisation token is required in the header of subsequent API calls and so none can be made without this token. Previous versions of Postman do not have this restriction which appears to have been introduced in version 7. If I retrieve the token using curl then Postman is quite happy send it in subsequent request headers. That is, Postman is restricting the size of response headers but appears able to send request headers without restriction.

The token request is http POST:
https://iam.ap-southeast-1.myhuaweicloud.com/v3/auth/tokens

The body content contains the client credentials which I cannot expose here.

**Comment by codenirvana** [comment]
@mikej0hnst0n Thanks for reporting this issue. This is a regression introduced in Postman v7.25.1 and will be fixed as soon as possible.

Till then, please use the manual workaround detailed below to specify the maximum HTTP header size (in bytes)

The workaround includes passing an additional environment variable `NODE_OPTIONS=--max-http-header-size=16384` (16KB) to Postman while launching. Here’s how to do this in various platforms:

### MacOS
```
NODE_OPTIONS=--max-http-header-size=16384 /Applications/Postman.app/Contents/MacOS/Postman
```

### Linux _(use the path where Postman is installed)_
```
NODE_OPTIONS=--max-http-header-size=16384 /path/to/Postman/Postman
```

### Windows
```
set NODE_OPTIONS=--max-http-header-size=16384
C:\users\<username>\AppData\local\Postman\Postman.exe
```

**Comment by fkennedy** [comment]
I'm also getting this error. Using the manual workaround above did not work.

**Comment by thiagoeh** [comment]
Same error, same API endpoint. Was able to solve using the workaround.
My computer runs Windows.

Instead of setting the variable temporarily, I used:
`SETX NODE_OPTIONS "--max-http-header-size=16384"`
at the Command Prompt

@fkennedy

**Comment by scottchen625** [comment]
My Mac runs 10.14.6. Running the `NODE_OPTIONS=--max-http-header-size=16384 /Applications/Postman.app/Contents/MacOS/Postman` command in terminal didn't solve the issue for me (it bugged out with an Electron errror). Neither did the advice in this post https://github.com/postmanlabs/newman/issues/2210

Any help is appreciated

Edit: Somehow updating to 10.15 and running the NODE_OPTIONS command again solved the issue

**Comment by dinhthaidaica** [comment]
any update on this? I'm facing this issue with MacOS 10.15.5 and Postman 7.28.0
running `NODE_OPTIONS=--max-http-header-size=16384 /Applications/Postman.app/Contents/MacOS/Postman` throwed me an error message: `(electron) 'getName function' is deprecated...`

**Comment by irondsd** [comment]
Is there any permanent fix for this issue? Running the command fixes the issue temporarily. If I run postman normally, the issue persists.

**Comment by dongweiq** [comment]
> 这事有进一步更新吗？我在MacOS 10.15.5和Postman 7.28.0
> 运行时遇到此问题，并向`NODE_OPTIONS=--max-http-header-size=16384 /Applications/Postman.app/Contents/MacOS/Postman`我抛出错误消息：`(electron) 'getName function' is deprecated...`

me too

**Comment by akangcupez** [comment]
I was succeed by deleting corresponding cookies from Cookies Manager below the Save button

**Comment by msn60** [comment]
I have the same issue and I could not fix this issue by settings max-http-header-size. 
I use Windows 10 and the version of Postman is 7.30.1 but this issue remains and I can not get the response of some POST requests. 
I want to ask how can I fix it or when Postman contributors fix it.

**Comment by cranium44** [comment]
I have the same issue and i am using Postman 7.31.1, Ubuntu 20.04 LTS. I get the error when making POST requests to certain third party APIs.

**Comment by mikej0hnst0n** [comment]
Just to be clear on the solution which works for me, permanently. I have set SYSTEM environment variable NODE_OPTIONS to --max-http-header-size=16384. That's on Windows 10.
![nodeopt](https://user-images.githubusercontent.com/66929227/91977318-15aec600-ed55-11ea-9263-57d06559daa8.JPG)

**Comment by mccannt** [comment]
We are still looking into this, I believe we the resolution identified and it shouldn't be too much longer before we have this resolved for everyone.

**Comment by daviddayanto** [comment]
Any update since 9 days ago has anyone elese figured this out?

**Comment by mccannt** [comment]
@daviddayanto Apologies. These things can take time to resolve fully and properly, so please hang tight.
Have you tried the workaround from above?

**Comment by msn60** [comment]
> Just to be clear on the solution which works for me, permanently. I have set SYSTEM environment variable NODE_OPTIONS to --max-http-header-size=16384. That's on Windows 10.
> ![nodeopt](https://user-images.githubusercontent.com/66929227/91977318-15aec600-ed55-11ea-9263-57d06559daa8.JPG)

Thanks. With this definition in system variable, it works

**Comment by PCurd** [comment]
As a note, we had a ~40KB header doing the same thing so we had to increase the environment variable a fair bit but it still worked with

```
set NODE_OPTIONS=--max-http-header-size=81920
C:\users\<username>\AppData\local\Postman\Postman.exe
```

It would be helpful if Postman could at least tell us how big the header that overflowed is - we had to find this value with trial and error!

**Comment by hellotorax** [comment]
I am still unable to use it with above solution. Getting Error:Parser Error:Header OVERFLOW. Can someone help

**Comment by sangampandey** [comment]
The only way for this to work on a mac is to quit your Postman and then try this command. It works perfectly fine and then launches the postman

`NODE_OPTIONS=--max-http-header-size=16384 /Applications/Postman.app/Contents/MacOS/Postman`

NOTE: The only issue is you have to run postman through this method always

**Comment by leen-x** [comment]
```sh
cp /Applications/Postman.app/Contents/MacOS/Postman /Applications/Postman.app/Contents/MacOS/Postman.real

sudo printf '#!/bin/bash \n NODE_OPTIONS=--max-http-header-size=16384 /Applications/Postman.app/Contents/MacOS/Postman.real' > /Applications/Postman.app/Contents/MacOS/Postman

sudo chmod u+x /Applications/Postman.app/Contents/MacOS/Postman
```
try ^-^

**Comment by space4yyy** [comment]
> ```shell
> cp /Applications/Postman.app/Contents/MacOS/Postman /Applications/Postman.app/Contents/MacOS/Postman.real
> 
> sudo printf '#!/bin/bash \n NODE_OPTIONS=--max-http-header-size=16384 /Applications/Postman.app/Contents/MacOS/Postman.real' > /Applications/Postman.app/Contents/MacOS/Postman
> 
> sudo chmod u+x /Applications/Postman.app/Contents/MacOS/Postman
> ```
> 
> try ^-^

it works

**Comment by 4selin** [comment]
the same for me

workaround works
I'm on Linux
`export NODE_OPTIONS=--max-http-header-size=16384`

**Comment by icnocop** [comment]
Setting `NODE_OPTIONS=--max-http-header-size=32767` broke Visual Studio 2017's extension updates and causes it to crash.

See https://superuser.com/questions/1397470/visual-studio-crashes-when-opening-project

**Comment by JAlonsoHdz** [comment]
> The only way for this to work on a mac is to quit your Postman and then try this command. It works perfectly fine and then launches the postman
> 
> `NODE_OPTIONS=--max-http-header-size=16384 /Applications/Postman.app/Contents/MacOS/Postman`
> 
> NOTE: The only issue is you have to run postman through this method always

Thanks sir, this worked for me.

**Comment by OnlyOneBDS** [comment]
I'm also getting this error using Postman v8.0.4

**Comment by VaishnaviChallaPostman** [comment]
```
I'm using the Postman desktop app because I'm testing localhost.

The command below did not make the error message disappear:
set NODE_OPTIONS=--max-http-header-size=16384 C:\Users\Acer\AppData\local\Postman\Postman.exe

Can you please advise how to proceed?
```

This is a request from a user, could you please suggest any possible alternative workaround for such case! Thanks :)

**Comment by swiknaba** [comment]
Error is also present in version 8.0.7 on macOS. My headers contain nothing besides the default stuff when requesting JSON from a REST API plus a JWT token. Starting the app via `NODE_OPTIONS=--max-http-header-size=16384 /Applications/Postman.app/Contents/MacOS/Postman` does work.

**Comment by chung-nd-kv** [comment]
I'm also getting this error using Postman v8.0.9 on Window 10 and the workaround works for me

**Comment by mccannt** [comment]
Thanks for the feedback everyone. We are actively looking at this now, so stay tuned :)

**Comment by numaanashraf** [comment]
We've included a fix for this issue in v8.4.0. Do let us know if the issue has resolved at your end.
